### PR TITLE
chore: move build tool to vite

### PR DESCRIPTION
**Pull request by prix0007**
Moving build tool from react-scripts to vite.
Reasons:
- Improved Performance.
Better build speeds
Better support

Reasons to deprecate react-scripts:
- Slower to build and no longer maintainer by the react team.

### CODE REVIEW

File: plugin/package.json [hunk 1]
```diff
@@ -47,12 +48,13 @@
...
     "eslint-plugin-react": "^7.32.2",
-    "typescript": "^4.9.5"
+    "typescript": "^4.9.5",
+    "vite": "^4.4.9"
   },
   "scripts": {
-    "start": "DISABLE_ESLINT_PLUGIN=true react-scripts start",
-    "build": "DISABLE_ESLINT_PLUGIN=true react-scripts build",
-    "serve": "pnpx serve -s build",
+    "start": "vite",
```
Comment (EdgarBarrantes) [inline plugin/package.json:55]: To avoid having to change configs in other places, add: `--port 3000` here.

File: plugin/package.json [hunk 2]
```diff
@@ -47,12 +48,13 @@
...
+    "build": "vite build",
+    "serve": "vite preview",
```
Comment (EdgarBarrantes) [inline plugin/package.json:57]: And here (`--port 3000`)

**Review by EdgarBarrantes — CHANGES_REQUESTED** [review]

**Comment by EdgarBarrantes** [comment]
Work great. Thanks for this. After comments and conflicts it should be good to go.

File: plugin/package.json (hunk 2 above)
Comment (prix0007) [reply plugin/package.json:57]: added port in the vite config.

Comment (prix0007) [reply plugin/package.json]: added port in serve command also.

File: plugin/package.json (hunk 1 above)
Comment (prix0007) [reply plugin/package.json:55]: yup added for dev server in the vite config. 
ATM both will run at 3000

**Review by EdgarBarrantes — APPROVED** [review]
It seems to run well, thanks a lot for this PR! :shipit:

**Pull request merged**